File: C++/operator/comparison.cpp
using namespace std;
#include <iostream>

void numberCompare();
void stringCompare();
void charCompare();
void decimalCompare();

int main() {
  numberCompare();
  stringCompare();
  charCompare();
  decimalCompare();
}

void numberCompare() {
  int x = 5;
  int y = 3;

  std::cout << "*** 0 is false; 1 is true;" << std::endl;
  std::cout << "5 == 3: " << (x == y) << std::endl;
  std::cout << "5 != 3: " << (x != y) << std::endl;
  std::cout << "5 > 3: " << (x > y) << std::endl;
  std::cout << "5 < 3: " << (x < y) << std::endl;
  std::cout << "5 >= 3: " << (x >= y) << std::endl;
  std::cout << "5 <= 3: " << (x <= y) << std::endl;

  std::cout << "\n" << std::endl;
}

void stringCompare() {
  string str1 = "Hello";
  string str2 = "hello";

  std::cout << "*** 0 is false; 1 is true;" << std::endl;
  std::cout << "Hello == hello: " << (str1 == str2) << std::endl;
  std::cout << "Hello != hello: " << (str1 != str2) << std::endl;
  std::cout << "Hello > hello: " << (str1 > str2) << std::endl;
  std::cout << "Hello < hello: " << (str1 < str2) << std::endl;
  std::cout << "Hello >= hello: " << (str1 >= str2) << std::endl;
  std::cout << "Hello <= hello: " << (str1 <= str2) << std::endl;

  std::cout << "\n" << std::endl;
}

void charCompare() {
  char char1 = 'a';
  char char2 = 'b';

  std::cout << "*** 0 is false; 1 is true;" << std::endl;
  std::cout << "char1 == char2: " << (char1 == char2) << std::endl;
  std::cout << "char1 != char2: " << (char1 != char2) << std::endl;
  std::cout << "char1 > char2: " << (char1 > char2) << std::endl;
  std::cout << "char1 < char2: " << (char1 < char2) << std::endl;
  std::cout << "char1 >= char2: " << (char1 >= char2) << std::endl;
  std::cout << "char1 <= char2: " << (char1 <= char2) << std::endl;

  std::cout << "\n" << std::endl;
}

void decimalCompare() {
  double x = 5.5;
  double y = 3.2;

  std::cout << "*** 0 is false; 1 is true;" << std::endl;
  std::cout << "5.5 == 3.2: " << (x == y) << std::endl;
  std::cout << "5.5 != 3.2: " << (x != y) << std::endl;
  std::cout << "5.5 > 3.2: " << (x > y) << std::endl;
  std::cout << "5.5 < 3.2: " << (x < y) << std::endl;
  std::cout << "5.5 >= 3.2: " << (x >= y) << std::endl;
  std::cout << "5.5 <= 3.2: " << (x <= y) << std::endl;

  std::cout << "\n" << std::endl;
}

File: JavaScript/operator/comparison.js
function NumberCompare() {
  console.log("******** Number Compare ********");

  let x = 10;
  let y = 5.3;
  console.log("x == y is ", x == y);
  console.log("x === y is ", x === y);
  console.log("x != y is ", x != y);
  console.log("x !== y is ", x !== y);
  console.log("x > y is ", x > y);
  console.log("x < y is ", x < y);
  console.log("x >= y is ", x >= y);
  console.log("x <= y is ", x <= y);
  console.log("\n");
}

function StringCompare() {
  console.log("********String Compare ********");

  let text1 = "Hello";
  let text2 = "hello";
  console.log("Hello == hello is ", text1 == text2);
  console.log("Hello === hello is ", text1 === text2);
  console.log("Hello != hello is ", text1 != text2);
  console.log("Hello !== hello is ", text1 !== text2);
  console.log("Hello > hello is ", text1 > text2);
  console.log("Hello < hello is ", text1 < text2);
  console.log("Hello >= hello is ", text1 >= text2);
  console.log("Hello <= hello is ", text1 <= text2);
  console.log("\n");

  let numtText1 = "20";
  let numtText2 = "5";
  console.log("20 == 5 is ", numtText1 == numtText2);
  console.log("20 === 5 is ", numtText1 === numtText2);
  console.log("20 != 5 is ", numtText1 != numtText2);
  console.log("20 !== 5 is ", numtText1 !== numtText2);
  console.log("20 > 5 is ", numtText1 > numtText2);
  console.log("20 < 5 is ", numtText1 < numtText2);
  console.log("20 >= 5 is ", numtText1 >= numtText2);
  console.log("20 <= 5 is ", numtText1 <= numtText2);
  console.log("\n");

  let char1 = "A";
  let char2 = "b";
  console.log("A == b is ", char1 == char2);
  console.log("A === b is ", char1 === char2);
  console.log("A != b is ", char1 != char2);
  console.log("A !== b is ", char1 !== char2);
  console.log("A > b is ", char1 > char2);
  console.log("A < b is ", char1 < char2);
  console.log("A >= b is ", char1 >= char2);
  console.log("A <= b is ", char1 <= char2);
  console.log("\n");
}

function MixCompare() {
  let num = 1;
  let str = "1";
  console.log("num == string is ", num == str);
  console.log("num === string is ", num === str);
  console.log("num != string is ", num != str);
  console.log("num !== string is ", num !== str);
  console.log("num > string is ", num > str);
  console.log("num < string is ", num < str);
  console.log("num >= string is ", num >= str);
  console.log("num <= string is ", num <= str);
  console.log("\n");
}

NumberCompare();
StringCompare();
MixCompare();
